tests/python/MypyBearTest.py

```diff
@@ -0,0 +1,124 @@
+                            ["print(123)"], valid=True)
+        self.check_validity(self.uut,
+                            ["print 123"], valid=False)
+        self.section.append(Setting('python_version', '2.7'))
+        self.check_validity(self.uut,
+                            ["print 123"], valid=True)
+
+    def test_bad_language(self):
+        self.section.append(Setting('language', 'Piet'))
+        self.check_validity(self.uut,
+                            ["1 + 1"], valid=True)
+        while not self.queue.empty():
+            message = self.queue.get()
+            msg = ('Language needs to be "Python", "Python 2" or '
+                   '"Python 3". Assuming Python 3.')
+            if message.message == msg:
+                break
+        else:
+            assert False, 'Message not found'
+
+    def test_check_untyped_function_bodies(self):
+        source = dedent("""
+            def foo():
+                return 1 + "abc"
+        """).splitlines()
+        self.check_validity(self.uut, source, valid=True)
+        self.section.append(Setting('check_untyped_function_bodies', 'true'))
+        self.check_validity(self.uut, source, valid=False)
+
+    def test_allow_untyped_functions(self):
+        source = dedent("""
+            def foo():
+                pass
+        """).splitlines()
+        self.check_validity(self.uut, source, valid=True)
+        self.section.append(Setting('allow_untyped_functions', 'false'))
+        self.check_validity(self.uut, source, valid=False)
+
+    def test_allow_untyped_calls(self):
+        source = dedent("""
+            def foo():
+                pass
+
+            foo()
+        """).splitlines()
+        self.check_validity(self.uut, source, valid=True)
+        self.section.append(Setting('allow_untyped_calls', 'false'))
+        self.check_validity(self.uut, source, valid=False)
+
+    def test_strict_optional(self):
+        source = dedent("""
+            from typing import Optional, List
+
+            def read_data_file(path: Optional[str]) -> List[str]:
+                with open(path) as f:  # Error
+                    return f.read().split(',')
+        """).splitlines()
+        self.check_validity(self.uut, source, valid=True)
+        self.section.append(Setting('strict_optional', 'true'))
+        self.check_validity(self.uut, source, valid=False)
+
+    def test_discarded_note(self):
+        source = dedent("""
+            def f() -> None:
+                1 + "a"
+        """).splitlines()
+        prepared = prepare_file(source, filename=None, create_tempfile=True)
+        with prepared as (file, fname):
+            results = [
+                Result.from_values(
+                    message=(
+                        'Unsupported operand types for + ("int" and "str")'),
+                    file=fname,
+                    line=3,
+                    origin=self.uut,
+                    severity=RESULT_SEVERITY.MAJOR,
+                )
```

Comment:
Have we changed to this style of bracket closing now ?
Earlier this would have been in the above line right ?